src/interceptors/ClientRequest/index.ts

```diff
@@ -39,17 +39,11 @@ function handleClientRequest(
     observer,
     resolver,
   })
-
-  debug('patching native http.ClientRequest...')
-
-  // Only http.ClientRequest is overridden because https uses http
-  // @ts-ignore
-  http.ClientRequest = ClientRequestOverride
```

Comment:
It's worth mentioning that by removing the `ClientRequest` patching we can no longer spy on the requests that are made by creating a direct `http.ClientRequest` instance. All the use-cases pass at the moment because we assert against `http.get`/`http.request` Node.js API that encapsulates `ClientRequest`. We stub those `get`/`request` functions, and that's how we are able to spy on any made request. 

I'd like to research if the direct usage of `ClientRequest` is a thing, and whether we should cover for it. 

My main concern is that there may be third-party libraries that use `ClientRequest` directly.

Reply:
> This object is created _internally_ and returned from `http.request()`
> —[Node.js documentation of `http.ClientRequest`](https://nodejs.org/api/http.html#http_class_http_clientrequest)

I think there is no use case to create the `http.ClientRequest` instance directly. Looks like we may omit to patch the `ClientRequest` directly.

Reply:
Yes, that might be a valid use case. But if thats needed then patching should be done during the initialisation (`server.listen()`) time itself alongside where we patch `http.get / http.request` I think..